### DISCUSSION TITLE
C/C++ binaries

### DISCUSSION THREAD

**Discussion by IshaB-07** [microsoft/vscode-cpptools#10031] [Q&A]
I am constantly getting this error  "The C/C++ extension doesn't match the system". I am unable to compile my program and run makefile. Please help me with the same.

<img width="503" alt="image" src="https://user-images.githubusercontent.com/113483305/196488602-90b19255-d4bd-4566-a49b-e85634794b72.png">

**Comment by Colengms** [comment]
Hi @IshaB-07 .  What system and architecture are you running on?  Are you using remoting (i.e. the Remote SSH extension)?  Are you installing via the Marketplace UI in VS Code, or are you manually installing a VSIX?

I'm not aware of any known issues causing the wrong extension architecture to be install via the Marketplace UI.  If you're installing using a VSIX, ensure you're installing the correct one, as the VSIX's for cpptools are platform specific.

I'm aware of [this issue](https://github.com/microsoft/vscode-remote-release/issues/7019#issuecomment-1207712796) which will install the wrong binaries on ARM64 Windows via Remote SSH, but that should not result in the issue you are seeing, as that also installs the wrong binaries for VS Code itself.

**Reply by IshaB-07** [reply]
I am using a Dell Laptop.
I downloaded it from this link : https://code.visualstudio.com/
It has 64-bit Operating System.

**Reply by Colengms** [reply]
Hi @IshaB-07 .  It sounds like you're using x64 Windows?  The link you referred to would be for downloading VS Code itself.  But, how did you install the C/C++ Extension into VS Code?  There would seem to be something unusual about how you installed the C/C++ Extension, that led to the issue.

Installing using the Marketplace UI within VS Code, should install the the correct VSIX.

![image](https://user-images.githubusercontent.com/49173979/196557162-34d966b9-4b28-4083-98c7-48c2eb5eeaf2.png)

Or, are did you install it some other way?

**Reply by IshaB-07** [reply]
Yes, I downloaded the extensions from the "EXTENSIONS" tab. 
However, I figured it out.

So, I have to compile a makefile but I am getting a "Make not found" error.

<img width="338" alt="image" src="https://user-images.githubusercontent.com/113483305/196565868-e4ef08be-f87c-4e17-a3ae-ca2d357cbdf6.png">

I have downloaded Makefile tools but it's still not working.

**Reply by Colengms** [reply]
The error, `The C/C++ extension doesn't match the system`, refers to native binaries used by the C/C++ extension, which _should_ match your system when installed using the Marketplace UI.  If the Marketplace UI is deploying an incorrect platform-specific extension, that's a bug in VS Code (or the marketplace itself), and it would be good for us to figure that out and fix it so it doesn't impact more users.  I think the issue you are seeing with Makefile Tools is separate and might be fixed by [installing Make](https://www.gnu.org/software/make/).

If you are still seeing the `doesn't match the system` error, and the C/C++ extension appears to be running properly, it could be that the x86 binary was installed on your x64 system, which would actually work properly.  There is a tab in the Task Manager, in the Detail view (and which might require enabling) that displays the architectures each process is compiled for.  If you see a `cpptools.exe` and/or `cpptools-srv.exe` on that list, could you tell me if they indicate x64 or x86?  You might also want to check if the executable for VS Code is itself x86 or x64.

You might also try uninstalling the C/C++ Extension and completely removing it from the extensions directory, as VS Code will cache it there and would reuse it when if you attempt to reinstall. On Windows, that would normally be here: `C:\Users\<your profile name>\.vscode\extensions`.  Then, try reinstalling the C/C++ Extension from the marketplace.  Perhaps that would download the correct version this time.

**Reply by IshaB-07** [reply]
I have installed makefile but it still won't work
